### PR TITLE
Update test_preproc.py

### PR DESCRIPTION
**Pull request by leoscholl**
trying to address #240 and #239

### CODE REVIEW

File: aopy/preproc/base.py [hunk 1]
```diff
@@ -331,12 +332,15 @@ def trial_align_data(data, trigger_times, time_before, time_after, samplerate):
         if np.isnan(t0):
             continue
         # sub = subvec(data, t0, n_samples, samplerate)
-        trial_data = np.empty((n_samples,data.shape[1]))
+        trial_data = np.zeros((n_samples,data.shape[1]))*np.nan
         idx_start = int(np.floor(t0*samplerate))
         idx_end = min(data.shape[0], idx_start+n_samples)
-        trial_data[:idx_end-idx_start,:] = data[idx_start:idx_end,:]
+        if idx_start < 0:
+            trial_data[-idx_start:idx_end-idx_start] = data[:idx_end,:]
```
Comment (mmadduri) [inline aopy/preproc/base.py:339]: will data always have idx_end data points?

File: tests/test_preproc.py [hunk 2]
```diff
@@ -317,24 +317,49 @@ def test_trial_align_events_to_list_of_tuples(self):
         np.testing.assert_allclose(expected_aligned_times, trial_aligned_times)
         
     def test_trial_align_data(self):
+        
+        # Test simple case with 0 time_before
         data = np.arange(100)
         samplerate = 1
         time_before = 0
         time_after = 10
         trigger_times = np.array([5, 55])
```
Comment (mmadduri) [inline tests/test_preproc.py:326]: could you add a comment about the 5 and 55?

File: tests/test_preproc.py [hunk 3]
```diff
@@ -317,24 +317,49 @@ def test_trial_align_events_to_list_of_tuples(self):
...
-        self.assertTrue(np.allclose(trial_aligned[1], np.arange(55, 65)))
+        np.testing.assert_allclose(np.squeeze(trial_aligned[0]), np.arange(5, 15))
+        np.testing.assert_allclose(np.squeeze(trial_aligned[1]), np.arange(55, 65))
+        
+        # Test with nonzero time_before
+        time_before = 2
+        trial_aligned = trial_align_data(data, trigger_times, time_before, time_after, samplerate)
+        self.assertEqual(len(trial_aligned), len(trigger_times))
+        np.testing.assert_allclose(np.squeeze(trial_aligned[0]), np.arange(3, 15))
+        np.testing.assert_allclose(np.squeeze(trial_aligned[1]), np.arange(53, 65))
```
Comment (mmadduri) [inline tests/test_preproc.py:337]: actually comments about all these numbers too

**Comment by mmadduri** [comment]
@leoscholl just a couple of questions for you

File: aopy/preproc/base.py (hunk 1 above)
Comment (leoscholl) [reply aopy/preproc/base.py:339]: yes because of `min(data.shape[0], ...)` a couple lines above

**Review by mmadduri — APPROVED** [review]
like the comments!

**Pull request merged**